js/models/questionModel.js

```diff
@@ -297,12 +294,12 @@ class QuestionModel extends ComponentModel {
             title,
             _classes,
             body: !isFinal
-              ? _feedback._partlyCorrect.notFinal
-              : _feedback._partlyCorrect.final
+              ? _feedback._partlyCorrect.notFinal || _feedback._incorrect?.notFinal || ''
+              : _feedback._partlyCorrect.final || _feedback._incorrect?.final || ''
```

Comment:
? _feedback._partlyCorrect?.notFinal || _feedback._incorrect?.notFinal || ''
: _feedback._partlyCorrect?.final || _feedback._incorrect?.final || ''

Reply:
Is there a difference in your suggestion to what is already in there? I can't see it but could easily be missing it 😂

Reply:
yes the suggestion of by adding the optional chaining operator with adding the extra `?` beside _feedback._partlyCorrect. In the same way you have added the operator beside the `_feedback._incorrect` object. It ensures that the _feedback._partlyCorrect object exists before trying to retrieve the value of the `notFinal` or `final` property.  If the object accessed or function called is undefined or null, it returns undefined instead of throwing an error. I added this suggestion to avoid an error being thrown should _feedback._partlyCorrect be null or undefined.

I have also noticed the following code in the `questionModel`:
`case 'partlyCorrect': {
        if (typeof _feedback._partlyCorrect === 'object') {
          // old style
          return {
            // add higher values
            altTitle,
            title,
            _classes,
            body: !isFinal
              ? _feedback._partlyCorrect.notFinal
              : _feedback._partlyCorrect.final
          };
        }
        // new style
        const feedbackPartlyCorrect = !isFinal ? _feedback._partlyCorrectNotFinal : _feedback._partlyCorrectFinal;`

this is open to debate, but we might want to also consider adding `_feedback._partlyCorrectNotFinal`  and `_feedback._partlyCorrectFinal` into the equation so doing something like this:

`? _feedback._partlyCorrect?.notFinal || _feedback._partlyCorrectNotFinal || _feedback._incorrect?.notFinal || ''
              : _feedback._partlyCorrect?.final || _feedback._partlyCorrectFinal` || _feedback._incorrect?.final || ''`

Reply:
Added that chaining operator. Good catch 👍